cli/flox/src/commands/init/go.rs

```diff
@@ -0,0 +1,481 @@
+                version: ProvidedVersion::Compatible {
+                    requested: None,
+                    compatible: ProvidedPackage::new("go", vec!["go"], "1.22.1"),
+                },
+            }),
+        };
+        assert!(go.should_run(Path::new("")).unwrap());
+    }
+
+    #[test]
+    fn test_should_run_returns_true_on_valid_workspace() {
+        let mut go = Go {
+            module_system: GoModuleSystemKind::Workspace(GoWorkspaceSystem {
+                version: ProvidedVersion::Compatible {
+                    requested: None,
+                    compatible: ProvidedPackage::new("go", vec!["go"], "1.22.1"),
+                },
+            }),
+        };
+        assert!(go.should_run(Path::new("")).unwrap());
+    }
+
+    #[test]
+    fn test_should_run_returns_false_on_none_system() {
+        let mut go = Go {
+            module_system: GoModuleSystemKind::None,
+        };
+        assert!(!go.should_run(Path::new("")).unwrap());
+    }
+
+    /*
+    #[test]
+    fn test_module_returns_error_on_invalid_content() {
+        let (flox, _) = flox_instance_with_global_lock();
+        let content = indoc! {r#"
+                go invalid!
+            "#};
+
+        let module = GoModuleSystem::try_new_from_content(content, &flox);
+
+        assert!(module.is_err());
+    }
+
+    #[test]
+    fn test_module_returns_error_on_invalid_content_version() {
+        let (flox, _) = flox_instance_with_global_lock();
+        let content = indoc! {r#"
+                // invalid go version
+                go invalid!
+            "#};
+
+        let module = GoModuleSystem::try_new_from_content(content, &flox);
+
+        assert!(module.is_err());
+    }
+
+    #[test]
+    fn test_workspace_returns_error_on_invalid_content() {
+        let (flox, _) = flox_instance_with_global_lock();
+        let content = indoc! {r#"
+                go invalid!
+            "#};
+
+        let workspace = GoWorkspaceSystem::try_new_from_content(content, &flox);
+
+        assert!(workspace.is_err());
+    }
+
+    #[test]
+    fn test_workspace_returns_error_on_invalid_content_version() {
+        let (flox, _) = flox_instance_with_global_lock();
+        let content = indoc! {r#"
+                // invalid go version
+                go invalid!
+            "#};
+
+        let workspace = GoWorkspaceSystem::try_new_from_content(content, &flox);
+
+        assert!(workspace.is_err());
+    }
+    */
+
+    #[test]
+    fn test_go_version_parsing_fails_with_invalid_version() {
+        let (flox, _) = flox_instance_with_global_lock();
+        let content = indoc! {r#"
+                // ivalid go version
+                go invalid
+            "#};
+
+        let version = ProvidedVersion::from_module_system_content(content, &flox);
+
+        assert!(!version.is_err());
+    }
+
+    #[test]
+    fn test_go_version_is_compatible() {
+        let (flox, _) = flox_instance_with_global_lock();
```

Comment:
With `_`, the temp dir will get dropped, so then subsequent operations will fail
```suggestion
        let (flox, _temp_dir_handle) = flox_instance_with_global_lock();
```

Reply:
I see, will change it for all instances :+1

Reply:
This solves my previous comment too :slightly_smiling_face: